packages/terra-search-field/src/SearchField.jsx

```diff
@@ -63,11 +74,17 @@ class SearchField extends React.Component {
   handleTextChange(event) {
     this.setState({ searchText: event.target.value });
 
-    if (!this.searchTimeout) {
+    if (!this.searchTimeout && !this.props.disableAutoSearch) {
       this.searchTimeout = setTimeout(this.handleSearch, this.props.searchDelay);
     }
   }
 
+  handleKeyDown(event) {
```

Comment:
I feel like this logic should move into the handleTextChange function.  and the searchTimeout logic should be broken into a resetSearchTimeout function.

Reply:
That being if we keep this,  handleKeyDown should be updated to something a bit more descriptive, like handleSearch

Reply:
handleTextChange is only used for when the value inside the search field actually does change, handling the ENTER key does not trigger the onChange event as the value stays the same.  Therefore, added this function to handle the keyPress of ENTER.
We already have a handleSearch function that both handleKeyDown and handleTextChange both call so I think handleKeyDown is descriptive and makes the most sense in that it is handing the down press of the ENTER key